lua/telescope/builtin/git.lua

```diff
@@ -114,24 +114,43 @@ git.branches = function(opts)
 end
 
 git.status = function(opts)
-  local output = utils.get_os_command_output{ 'git', 'status', '-s' }
+  local gen_new_finder = function()
+    local output = utils.get_os_command_output{ 'git', 'status', '-s' }
 
-  if table.getn(output) == 0 then
-    print('No changes found')
-    return
+    if table.getn(output) == 0 then
+      print('No changes found')
+      return
+    end
+
+    return finders.new_table {
+      results = output,
+      entry_maker = make_entry.gen_from_git_status(opts)
+    }
   end
 
+  local initial_finder = gen_new_finder()
+  if not initial_finder then return end
+
   pickers.new(opts, {
     prompt_title = 'Git Status',
-    finder = finders.new_table {
-      results = output,
-      entry_maker = make_entry.gen_from_git_status(opts)
-    },
+    finder = initial_finder,
     previewer = previewers.git_file_diff.new(opts),
     sorter = conf.file_sorter(opts),
     attach_mappings = function(_, map)
-      map('i', '<tab>', actions.git_staging_toggle)
-      map('n', '<tab>', actions.git_staging_toggle)
+      local toggle_staging = function(prompt_bufnr)
+        local selection = actions.get_selected_entry()
+
+        if selection.status:sub(2) == ' ' then
+          os.execute('git restore --staged ' .. selection.value)
```

Comment:
Does this work from other directories... just noticed this may not do that.

You can use `Job:new { ... }:sync()` to run a command in process.

Reply:
I know all about Job :) but good point